src/common/hosts-patches-table.js

```diff
@@ -104,100 +114,70 @@ class HostsPatchesTable extends Component {
     return Promise.all(map(
       keys(pools),
       installAllPatchesOnPool
-    )).then(this._refreshMissingPatches)
-  }
-
-  _refreshHostMissingPatches = host => (
-    getHostMissingPatches(host).then(patches => {
-      this.setState({
-        missingPatches: {
-          ...this.state.missingPatches,
-          [host.id]: patches.length
-        }
-      })
-    })
-  )
-
-  _installAllHostPatches = host => (
-    installAllHostPatches(host).then(() =>
-      this._refreshHostMissingPatches(host)
-    )
-  )
-
-  componentWillMount () {
-    this._refreshMissingPatches()
+    ))
   }
 
   componentDidMount () {
     // Force one Portal refresh.
     // Because Portal cannot see the container reference at first rendering.
     this.forceUpdate()
+    this.unsubscribeMissingPatches = this._subscribeMissingPatches()
   }
 
   componentWillReceiveProps (nextProps) {
-    forEach(nextProps.hosts, host => {
-      const { id } = host
-
-      if (this.state.missingPatches[id] !== undefined) {
-        return
-      }
-
-      this.setState({
-        missingPatches: {
-          ...this.state.missingPatches,
-          [id]: 0
-        }
-      })
+    if (nextProps.hosts !== this.props.hosts) {
+      this.unsubscribeMissingPatches()
+      this.unsubscribeMissingPatches = this._subscribeMissingPatches(nextProps.hosts)
```

Comment:
Calling `unsubscribeMissingPatches` if necessary and assigning it should be done in `_subscribeMissingPatches`.